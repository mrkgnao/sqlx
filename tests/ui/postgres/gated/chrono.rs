fn main() {
    let _ = sqlx::query!("select now()::date");

    let _ = sqlx::query!("select now()::time");

    let _ = sqlx::query!("select now()::timestamp");

    let _ = sqlx::query!("select now()::timestamptz");
}
