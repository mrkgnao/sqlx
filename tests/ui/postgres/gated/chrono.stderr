error: support for column type date (type OID 1082) at position 0 (name: Some("now")) requires optional feature `chrono`
 --> $DIR/chrono.rs:2:13
  |
2 |     let _ = sqlx::query!("select now()::date");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: support for column type time (type OID 1083) at position 0 (name: Some("now")) requires optional feature `chrono`
 --> $DIR/chrono.rs:4:13
  |
4 |     let _ = sqlx::query!("select now()::time");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: support for column type timestamp (type OID 1114) at position 0 (name: Some("now")) requires optional feature `chrono`
 --> $DIR/chrono.rs:6:13
  |
6 |     let _ = sqlx::query!("select now()::timestamp");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: support for column type timestamptz (type OID 1184) at position 0 (name: Some("now")) requires optional feature `chrono`
 --> $DIR/chrono.rs:8:13
  |
8 |     let _ = sqlx::query!("select now()::timestamptz");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: support for type date (type OID 1082) of param #1 requires optional feature `chrono`
  --> $DIR/chrono.rs:10:13
   |
10 |     let _ = sqlx::query!("select $1::date", ());
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: support for type time (type OID 1083) of param #1 requires optional feature `chrono`
  --> $DIR/chrono.rs:12:13
   |
12 |     let _ = sqlx::query!("select $1::time", ());
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: support for type timestamp (type OID 1114) of param #1 requires optional feature `chrono`
  --> $DIR/chrono.rs:14:13
   |
14 |     let _ = sqlx::query!("select $1::timestamp", ());
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: support for type timestamptz (type OID 1184) of param #1 requires optional feature `chrono`
  --> $DIR/chrono.rs:16:13
   |
16 |     let _ = sqlx::query!("select $1::timestamptz", ());
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
